openfe/protocols/openmm_utils/settings_validation.py

```diff
@@ -71,3 +71,95 @@ def get_simsteps(sim_length: unit.Quantity,
         raise ValueError(errmsg)
 
     return sim_steps
+
+
+def convert_steps_per_iteration(
+        simulation_settings,
+        integrator_settings,
+):
+    """Convert time per iteration to steps
+
+    Parameters
+    ----------
+        simulation_settings: MultiStateSimulationSettings
+    integrator_settings: IntegratorSettings
+
+    Returns
+    -------
+    steps_per_iteration : int
+      suitable for input to Integrator
+    """
+    # TODO: Check this is correct
+    tpi_fs = simulation_settings.time_per_iteration.to(unit.femtosecond).m
+    ts_fs = integrator_settings.timestep.to(unit.femtosecond).m
+    steps_per_iteration = int(round(tpi_fs / ts_fs))
+
+    return steps_per_iteration
+
+
+def convert_real_time_analysis_iterations(
+        simulation_settings,
+):
+    """Convert time units in Settings to various other units
+
+    Interally openmmtools uses various quantities with units of time,
+    steps, and iterations.
+
+    Our Settings objects instead have things defined in time (fs or ps).
+
+    This function generates suitable inputs for the openmmtools objects
+
+    Parameters
+    ----------
+    simulation_settings: MultiStateSimulationSettings
+
+    Returns
+    -------
+    real_time_analysis_iterations : int
+      suitable for input to online_analysis_interval
+    real_time_analysis_minimum_iterations : int
+      suitable for input to real_time_analysis_minimum_iterations
+    """
+    # TODO: Check this is correct
+    tpi_fs = simulation_settings.time_per_iteration.to(unit.femtosecond).m
+
+    # convert real_time_analysis time to interval
+    # rta_its must be number of MCMC iterations
+    # i.e. rta_fs / tpi_fs -> number of iterations
+    rta_fs = simulation_settings.real_time_analysis_interval.to(unit.femtosecond).m
+    rta_its = round(int(rta_fs / tpi_fs))
```

Comment:
I'm not sure I agree that we should be brute force converting things.

If the initial float values at the attosecond level don't divide through properly then it should raise a ValueError, forcing an `int` is going to make for a simulation with undefined behaviour (one that isn't even recorded in the output)